code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -161,56 +161,55 @@
 	var/chosen_target = pick(Targets)//Pick the remaining targets (if any) at random
 	return chosen_target
 
-/mob/living/simple_animal/hostile/CanAttack(atom/the_target)//Can we actually attack a possible target?
-	if(!the_target)
-		return 0
+// Please do not add one-off mob AIs here, but override this function for your mob
+/mob/living/simple_animal/hostile/CanAttack(atom/the_target)//Can we actually attack a possible target? 
+	if(isturf(the_target) || !the_target || the_target.type == /atom/movable/lighting_object) // bail out on invalids
```

Comment:
i really dislike this explicit type check

Reply:
It's faster than istype, we have no lighting subtypes and this proc is top3-5 by calls

Reply:
This is fine.